rpc/handlers.go

```diff
@@ -182,34 +182,35 @@ func adaptDeclareTransaction(t *core.DeclareTransaction) *Transaction {
 
 	if t.Version.Equal(new(felt.Felt).SetUint64(2)) {
 		txn.CompiledClassHash = nil // todo: add when we have support for Declare V2
-	} else if !t.Version.IsZero() && !t.Version.IsOne() {
-		panic("invalid invoke txn version")
 	}
+
 	return txn
 }
 
 func (h *Handler) blockByID(id *BlockID) (*core.Block, error) {
 	if id.Latest {
 		return h.bcReader.Head()
-	} else if id.Hash != nil {
+	}
```

Comment:
Let's make this a `switch`? The conditions we are testing are mutually exclusive, so I believe we should reflect that it code.

Reply:
done